app/src/models/repository.ts

```diff
@@ -72,11 +80,35 @@ export class Repository {
       this.workflowPreferences.forkContributionTarget,
       this.isTutorialRepository
     )
+
+    this.fetchUrl()
   }
 
   public get path(): string {
     return this.mainWorkTree.path
   }
+
+  public get url(): string | null {
+    return this._url
+  }
+
+  private fetchUrl(): void {
+    this._url = null
+
+    // cannot load remote from non-exisiting repository folder
+    // this is mostly needed for testing
+    // eslint-disable-next-line no-sync
+    if (!fs.existsSync(this.path)) {
+      return
+    }
```

Comment:
If we are sure that this can only occur during testing, it might be better to rewrite tests to use actual repositories instead of dummies.
E.g.:


https://github.com/desktop/desktop/blob/development/app%2Ftest%2Funit%2Fgit-store-cache-test.ts#L15-L22